src/ast.rs

```diff
@@ -1766,4 +1808,86 @@ event zeek_init() { for (v in vs) ; }
         let typ = db.typ(decl).unwrap();
         assert_debug_snapshot!(typ);
     }
+
+    #[test]
+    fn loop_vars_set_iteration_order() {
+        let mut db = TestDatabase::default();
+        let uri = Arc::new(Url::from_file_path("/x.zeek").unwrap());
+        db.add_file(
+            (*uri).clone(),
+            r#"
+global vs: set[string, count] = set("a", 1);
+event zeek_init() { for (v in vs) ; }
+                  "#,
+        );
+
+        let db = db.0;
+        let source = db.source(uri.clone()).unwrap();
+        let tree = db.parse(uri.clone()).unwrap();
+        let root = tree.root_node();
+
+        let vs = root
+            .named_descendant_for_position(Position::new(2, 31))
```

Comment:
This checks the type of `vs` in l.1820 while we'd be interested in the result of resolving `v`. Similar in the test below.